### PR TITLE
Add tooling list

### PR DESCRIPTION
**Pull request by kittysquee**
# [Review the dev and ops manuals (addressing the assumed knowledge)](https://github.com/barnardos/BAU/issues/3)

* Adds list of links to tooling docs
* Adds brief intro taken from top Google result/docs
* Added link to tooling page from README
* Ran prettier on README

Please feel free to change anything, this is just a start 😺

### CODE REVIEW

File: tooling.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# Tooling List
+
+This is a non-exhaustive list of some of the tooling that we use within the digital team. If you feel that something should be added or removed please submit a <abbr title="pull request">PR</abbr>.
```
Comment (jeddy3) [inline tooling.md]: For conciseness, shall we delete?

> If you feel that something should be added or removed please submit a <abbr title="pull request">PR</abbr>.

File: tooling.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# Tooling List
```
Comment (jeddy3) [inline tooling.md]: Will just "Tooling" suffice?

File: tooling.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+This is a non-exhaustive list of some of the tooling that we use within the digital team. If you feel that something should be added or removed please submit a <abbr title="pull request">PR</abbr>.
+
+[Git](https://www.atlassian.com/git/tutorials/what-is-git)
+
+> Git is a version control system for tracking changes in computer files and
+coordinating work on those files among multiple people.
+
+
```
Comment (jeddy3) [inline tooling.md]: Extra empty line

File: tooling.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+[GitHub](https://github.com/)
+
+> GitHub is a web-based hosting service for version control using git. It
+offers all of the distributed version control and source code management
+functionality of Git as well as adding its own features.
+
+
```
Comment (jeddy3) [inline tooling.md]: Extra empty line

File: tooling.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+[Circle CI](https://circleci.com/)
+
```
Comment (jeddy3) [inline tooling.md]: Can we add a description for this tool?

File: tooling.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+> PostgreSQL, often simply Postgres, is an object-relational database
+management system with an emphasis on extensibility and standards compliance
+
+[Docker](https://www.docker.com/)
+
+> Docker is an open platform for developers and sysadmins to build, ship, and
+run distributed applications.
+
+[BEM](http://getbem.com/)
```
Comment (jeddy3) [inline tooling.md]: Shall we bump this out to a `methodologies.md`?

File: tooling.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+> <abbr title="Block Element Modifier">BEM</abbr> is a methodology, that helps
+you to achieve reusable components and code sharing in the front-end.
+
+[8 point design](https://builttoadapt.io/intro-to-the-8-point-grid-system-d2573cde8632)
```
Comment (jeddy3) [inline tooling.md]: I think we can nuke this as it'll be captured in the interface design system.

File: tooling.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+
+[CHANGELOGs](http://keepachangelog.com/en/1.0.0/)
```
Comment (jeddy3) [inline tooling.md]: Shall we bump this out to a `methodologies.md`?

File: tooling.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+help lint your rote tasks in daily code review.
+> You can use Danger to codify your teams norms, leaving humans to think about
+harder problems.
+
+[Rubocop](https://github.com/bbatsov/rubocop)
+
+> RuboCop is a Ruby static code analyzer. Out of the box it will enforce many
+of the guidelines outlined in the community Ruby Style Guide.
+
+Linters
```
Comment (jeddy3) [inline tooling.md]: I think we can break these into individual tools (like Rubocop is) and remove the group "Linters".

File: tooling.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+- [eslint](https://eslint.org/)
+- [stylelint](https://github.com/stylelint/stylelint)
+- [remarklint](https://github.com/remarkjs/remark-lint)
+
+[Prettier](https://prettier.io/)
+
+> Prettier can be run in your editor on-save, in a pre-commit hook, or in <abbr title="Continuous Integration">CI</abbr> environments to ensure your codebase
+has a consistent style without devs ever having to think about it.
+
+Test driven development
```
Comment (jeddy3) [inline tooling.md]: Shall we bump this out to a `methodologies.md` and break out the individual tools?

File: tooling.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+> Gatsby is a static site generator built on React.
+
+[Storybook](https://storybook.js.org/)
+
+> Storybook is a UI development environment for your UI components. With it,
+you can visualize different states of your UI components and develop them
+interactively.
+
+> Storybook runs outside of your app. So you can develop UI components in
```
Comment (jeddy3) [inline tooling.md]: Shall we nuke this 2nd paragraph for consistency?

File: tooling.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+isolation without worrying about app specific dependencies and requirements.
+
+[Heroku](https://heroku.com)
+
+> Heroku is a cloud platform as a service supporting several programming
+languages that is used as a web application deployment model. To see our
```
Comment (jeddy3) [inline tooling.md]: Shall we remove this cross link? I think it changes the scope of this doc from just being a list to being an interwoven page, which is arguably harder to maintain.

File: tooling.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+[Circle CI](https://circleci.com/)
+
+
+[Yarn](https://yarnpkg.com/en/)
+
+> For managing JavaScript dependencies.
+
+[NPM](https://www.npmjs.com/)
+
+> <abbr title="node package manager">npm</abbr> is a package manager for
```
Comment (jeddy3) [inline tooling.md]: Can we add a trailing full stop?

File: tooling.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+
+[Ruby](https://www.ruby-lang.org/en/)
+
+> A dynamic, open source programming language with a focus on simplicity and
+productivity. It has an elegant syntax that is natural to read and easy to
+write.
+
+[Ruby on Rails](http://rubyonrails.org/)
+
+> Ruby on Rails, or Rails, is a server-side web application framework written
```
Comment (jeddy3) [inline tooling.md]: Can we add a trailing full stop?

File: tooling.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+corporations.
+
+[Redis](https://redis.io/)
+
+> Redis is an open source, in-memory data structure store, used as a database,
+cache and message broker.
+
+[PostgreSQL](https://www.postgresql.org/)
+
+> PostgreSQL, often simply Postgres, is an object-relational database
```
Comment (jeddy3) [inline tooling.md]: Can we add a trailing full stop?

File: tooling.md [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+
+> A changelog is a log or record of all notable changes made to a project. The
+project is often a website or software project, and the changelog usually
+includes records of changes such as bug fixes, new features, etc. Some open
+source projects include a changelog as one of the top level files in their
+distribution.
+
+[Danger](https://github.com/danger/danger)
+
+> Danger runs after your CI, automating your team's conventions surrounding
```
Comment (jeddy3) [inline tooling.md]: Shall we put these into a single paragraph for the sake of consistency?

File: README.md [hunk 17]
```diff
@@ -1,25 +1,30 @@
...
-You can call it [agile](agile/index.md), if you must. But please use a small 'a'. 
+You can call it [agile](agile/index.md), if you must. But please use a small 'a'.
 
 ### Some [styleguides](styleguides/index.md)
 
-We work in a variety of languages and with a variety of tools. These are [style guides](styleguides/index.md) 
+We work in a variety of languages and with a variety of tools. These are [style guides](styleguides/index.md)
 for languages, and also for things like commit messages and pull requests.
 
+### [Tooling List](tooling.md)
```
Comment (jeddy3) [inline README.md]: Will "Tooling" suffice here?

File: tooling.md [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+
+> Babel is a essentially an ECMAScript 6 to ECMAScript 5 transpiler. Or for the purposes of somewhat future-proofing this answer - ESNext to ES”current” transpiler. In short - it allows you to use language features “from the future”. It does so by transpiling your code to the currently supported version of JavaScript.
+
+[Circle CI](https://circleci.com/)
+
+> CircleCI is for Continuous Integration, a development practice which is being used by software teams allowing them to to build, test and deploy applications easier and quicker on multiple platforms.
+
+[Danger](https://github.com/danger/danger)
+
+> Danger runs after your CI, automating your team's conventions surrounding
```
Comment (jeddy3) [inline tooling.md]: Shall we merge these sentences into a single paragraph for consistency?

File: tooling.md [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+[Jekyll](https://jekyllrb.com/)
+
+> Jekyll is a simple static site generator built on Ruby.
+
+Linters
+
+> A linter or lint refers to tools that analyze source code to flag programming errors, bugs, stylistic errors, and suspicious constructs.
+
+> We use several linters at Barnardos:
+- [eslint](https://eslint.org/)
```
Comment (jeddy3) [inline tooling.md]: For consistency, shall we remove the "Linters" group and break the three tools below into individual entries?

File: tooling.md [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+
+[Ruby on Rails](http://rubyonrails.org/)
+
+> Ruby on Rails, or Rails, is a server-side web application framework written in Ruby under the <abbr title="Massachusetts Institute of Technology">MIT</abbr> License. Rails is a model–view–controller framework, providing default structures for a database, a web service, and web pages.
+
+[Storybook](https://storybook.js.org/)
+
+> Storybook is a UI development environment for your UI components. With it, you can visualize different states of your UI components and develop them interactively.
+
+> Storybook runs outside of your app. So you can develop UI components in isolation without worrying about app specific dependencies and requirements.
```
Comment (jeddy3) [inline tooling.md]: Shall we delete this second paragraph, so that there is consistently one paragraph per entry?

File: methodologies.md [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+
+[8 point design](https://builttoadapt.io/intro-to-the-8-point-grid-system-d2573cde8632)
+
+> A design system that uses multiples of 8 to define dimensions, padding, and margin of both block and inline elements.
+
+Test driven development
+
+> <abbr title="Test-driven development">TDD</abbr> is a software development process that relies on the repetition of a very short development cycle: first the developer writes an (initially failing) automated test case that defines a desired improvement or new function, then produces the minimum amount of code to pass that test, and finally refactors the new code to acceptable standards.
+
+> Some of the tools we use to test our code:
```
Comment (jeddy3) [inline methodologies.md]: Shall we move these tools out of this page and into the tools page, and then add individual entries for each?

File: methodologies.md [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+
+[CHANGELOGs](http://keepachangelog.com/en/1.0.0/)
+
+> A changelog is a log or record of all notable changes made to a project. The
+project is often a website or software project, and the changelog usually
+includes records of changes such as bug fixes, new features, etc. Some open
+source projects include a changelog as one of the top level files in their
+distribution.
+
+[8 point design](https://builttoadapt.io/intro-to-the-8-point-grid-system-d2573cde8632)
```
[the remark this answers is not in the document]
Comment (kittysquee) [reply methodologies.md]: ✅

**Pull request merged**